class.json-api-endpoints.php

```diff
@@ -2052,6 +2052,19 @@ function allows_unauthorized_requests( $origin, $complete_access_origins ) {
 		return 'GET' == $this->method || ( $this->allow_unauthorized_request && in_array( $origin, $complete_access_origins ) );
 	}
 
+	/**
+	 * Whether this endpoint allows site based authentication.
+	 *
+	 * @since 9.1.0
+	 *
+	 * @return bool true, if Jetpack blog token is used and `allow_jetpack_site_auth` is true,
+	 * false otherwise.
+	 */
+	public function allows_site_based_authentication() {
```

Comment:
A suggestion on naming.

What do you think of `accepts_site_based_authentication` or `check_site_based_authentication`... 

My point here is that this check is relative to the current request. Based on the endpoint configuration and on the authentication method used we determine if we should accept this request or not.

The "allows" word, as used in `allow_jetpack_site_auth` property sounds for me like something permanent, and not a per-request thing.

Am I being to picky?

Reply:
Not at all. I was also struggling with naming this method exactly because of the request/endpoint context.
If `accepts_site_based_authentication` sounds better to you I'm glad to update it :)